book/src/dev/rfcs/0006-stolon.md

```diff
@@ -0,0 +1,446 @@
+should have the following fields:
+
+| Field           | Value                                                   |
+|-----------------|---------------------------------------------------------|
+| `version`       | Current supported Zebra network version                 |
+| `services`      | `0`, no advertised services                             |
+| `timestamp`     | Local UNIX timestamp, truncated to 5-minute intervals   |
+| `address_recv`  | (`NODE_ADDR`, `0.0.0.0:8233`)                           |
+| `address_from`  | (`0`, `0.0.0.0:8233`)                                   |
+| `nonce`         | Random value                                            |
+| `user_agent`    | `""`, the empty string                                  |
+| `start_height`  | `0`                                                     |
+| `relay`         | `false`                                                 |
+
+The timestamp is chosen to be truncated to 5-minute intervals as a balance
+between two goals:
+
+* providing minimally-distinguishable data in the handshake;
+* not interfering with `zcashd`'s existing code.
+
+There is no specification of the network protocol, so when considering the
+latter goal the best we can do is to make inferences based on the current state
+of the `zcashd` source code.  Currently, `zcashd` only uses the handshake
+timestamp at the end of parsing the version message:
+```
+pfrom->nTimeOffset = timeWarning.AddTimeData(pfrom->addr, nTime, GetTime());
+```
+The `AddTimeData` method defined in `src/timedata.cpp` is a no-op as long as
+the difference between the specified timestamp and `zcashd`'s local time is less
+than a `TIMEDATA_WARNING_THRESHOLD` set to 10 minutes.  Otherwise, the data is fed
+into a process that attempts to detect when the local time is wrong.
+Truncating to the nearest 5 minutes (half of `TIMEDATA_WARNING_THRESHOLD`)
+attempts to try to stay within this interval, so as to not confuse `zcashd`.
+
+In general, truncation may not be sufficient to prevent an observer from
+obtaining distinguishing information, because of aliasing between the truncated
+data and the original signal.  For instance, continuously broadcasting the
+node's timestamp truncated to a 5-minute interval would not hide the node's
+clock skew, which would be revealed at the moment the node switched to the next
+truncation.  However, this is unlikely to be a problem in practice, because the
+number of samples an observer can obtain from a single node is likely to be
+small. And, unlike other approaches like adding random noise, truncation is
+simple, reliable, and deterministic, making it easy to test.
+
+## Deployment considerations and non-Tor relay fallback
+
+Unfortunately, Tor is difficult to use as a library and its ease of deployment
+varies by platform. We therefore need to consider how to support Tor
+integrations and to provide a fallback mechanism when Tor is not available.
```

Comment:
Does this open up an attack vector that allows an adversary to force connections to be downgraded to not over Tor?

Reply:
No, because network adversaries cannot control the deployment parameters.